xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -350,20 +458,80 @@ viewers.Comments = Class.create({
       this.initialLocation.insert({after: this.form});
     }
     this.form["XWiki.XWikiComments_replyto"].value = "";
-    this.form["XWiki.XWikiComments_comment"].value = "";
+    this.reloadEditor();
     this.cancelPreview(this.form);
   },
   /**
    * Registers a listener that watches for the insertion of the Comments tab and triggers the enhancements.
    * After that, the listener removes itself, since it is no longer needed.
    */
-  addTabLoadListener : function(event) {
-    var listener = function(event) {
+  addTabLoadListener: function (event) {
+    var listener = function (event) {
       if (event.memo.id == 'Comments') {
         this.startup();
       }
     }.bindAsEventListener(this);
     document.observe("xwiki:docextra:loaded", listener);
+  },
+  destroyEditor: function (selector, name) {
+    require(['jquery', 'xwiki-events-bridge'], function ($) {
+      $(document).trigger('xwiki:actions:cancel');
+      if ($(selector).size() > 0) {
+        const parent = $(selector).parent(".wysiwyg-field");
+        parent.empty();
+        $(document).trigger('xwiki:dom:updated', {'elements': parent.toArray()});
+      }
+    });
+  },
+  reloadEditor: function (options) {
+    var wfClass = '.wysiwyg-field';
+    options = options || {};
+    const commentNbr = options.commentNbr;
+    const callback = options.callback;
+    var name = 'XWiki.XWikiComments_comment';
+    if (commentNbr) {
+      name = 'XWiki.XWikiComments_' + commentNbr + '_comment';
+      wfClass = wfClass + '-' + commentNbr;
+    }
+
+    this.destroyEditor("[name='" + name + "']", name);
+
+    require(['jquery', 'xwiki-events-bridge'], function ($) {
+      if ($(".wysiwyg-field").size() > 0) {
+        $.post(new XWiki.Document().getURL("view") + '?' + $.param({
+          xpage: 'xpart',
+          vm: 'commentfield.vm',
+          number: commentNbr,
+          name: name,
+        }), function (data, status, jqXHR) {
+          function loadRequiredSkinExtensions(requiredSkinExtensions)
+          {
+            var existingSkinExtensions;
+            var getExistingSkinExtensions = function () {
+              return $('link, script').map(function () {
+                return $(this).attr('href') || $(this).attr('src');
+              }).get();
+            };
+            $('<div/>').html(requiredSkinExtensions).find('link, script').filter(function () {
+              if (!existingSkinExtensions) {
+                existingSkinExtensions = getExistingSkinExtensions();
+              }
+              var url = $(this).attr('href') || $(this).attr('src');
+              return existingSkinExtensions.indexOf(url) < 0;
+            }).appendTo('head');
+          }
+
+          const wf = $(wfClass);
+          wf.empty();
+          wf.append(data);
+          $(document).trigger('xwiki:dom:updated', {'elements': wf.toArray()});
+          loadRequiredSkinExtensions(jqXHR.getResponseHeader('X-XWIKI-HTML-HEAD'));
```

Comment:
If I'm right this is only needed for WYSIWYG editor, I don't know how it would behave for the text editor, but feels wrong to call it every time.

Reply:
The text editor could require, in theory, some skin extensions, so I don't see the problem with calling this for any editor. The ``X-XWIKI-HTML-HEAD`` header is simply not set or empty when the text editor is used I think, so there's no performance penalty.

Reply:
Ok I'll try to debug it to ensure it doesn't lead to an error with text editor (would explain why the callback isn't properly called to display preview in that case). But keeping it.

Reply:
Ok it's working properly.